application/src/main/java/org/togetherjava/tjbot/commands/tag/TagManageCommand.java

```diff
@@ -0,0 +1,233 @@
+        String content = event.getOption(CONTENT).getAsString();
+
+        if (tagSystem.exists(tagId)) {
+            event.reply("This tag already exists").setEphemeral(true).queue();
+
+            return;
+        }
+
+        tagSystem.put(tagId, content);
+
+        event
+            .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                .setTitle(SUCCESS)
+                .setTimestamp(LocalDateTime.now())
+                .setFooter(event.getUser().getAsTag())
+                .setDescription("Successfully created tag '" + tagId + "'")
+                .build())
+            .queue();
+    }
+
+    private void createIdTag(SlashCommandEvent event) {
+        String tagId = event.getOption(TAG_MINUS_ID).getAsString();
+        long messageId = event.getOption(MESSAGE_MINUS_ID).getAsLong();
+
+        if (tagSystem.exists(tagId)) {
+            event.reply("This tag already exists").setEphemeral(true).queue();
+
+            return;
+        }
+
+        event.getMessageChannel().retrieveMessageById(messageId).queue(message -> {
+            tagSystem.put(tagId, message.getContentRaw());
+
+            event
+                .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                    .setTitle(SUCCESS)
+                    .setTimestamp(LocalDateTime.now())
+                    .setFooter(event.getUser().getAsTag())
+                    .setDescription("Successfully created tag '" + tagId + "'")
+                    .build())
+                .queue();
+        }, failure -> {
+            if (failure instanceof ErrorResponseException ex
+                    && ex.getErrorResponse() == ErrorResponse.UNKNOWN_MESSAGE) {
+                event.reply("This message doesn't exist").setEphemeral(true).queue();
+            }
+        });
+    }
+
+    @Override
+    public void onButtonClick(ButtonClickEvent event, List<String> idArgs) {
+        String userId = idArgs.get(0);
+
+        if (event.getUser().getId().equals(userId)
+                || event.getMember().hasPermission(Permission.MESSAGE_MANAGE)) {
+            if (event.getButton().getLabel().equals("Of course!")
+                    && !event.getMember().hasPermission(Permission.MESSAGE_MANAGE)) {
+                String tagId = idArgs.get(1);
+
+                tagSystem.delete(tagId);
+
+                event.getMessage().delete().queue();
+
+                event
+                    .replyEmbeds(new EmbedBuilder().setColor(Color.GREEN)
+                        .setTitle(SUCCESS)
+                        .setTimestamp(LocalDateTime.now())
+                        .setFooter(event.getUser().getAsTag())
+                        .setDescription("Successfully deleted tag '" + tagId + "'")
+                        .build())
+                    .setEphemeral(true)
+                    .queue();
+            } else if (event.getButton().getLabel().equals("Abort")) {
+                event.getMessage().delete().queue();
+            }
+        } else {
```

Comment:
prefer early out. reverse the condition and have the rest without a big if. `if (!...) { ... return } ...`.

Reply:
done